### PR TITLE
feat(v3): optimized generated client code

### PR DESCRIPTION
**Pull request by vlastahajek**
Swagger generated code refactor according to the https://gist.github.com/rogpeppe/359e7465956e98bed4c5b9692b910a05:
 - Single method for each operation
 - Filtering operations only for those with JSON request body (POST and PATCH) and for those with a single JSON response with status code 2xx
 - Returning concrete response type, instead of a wrapper
 - Non 2xx responses result in error
 - Single request parameter - path, query, cookie and header params merged into single struct along with request body
 
Requires oapi-codegen with extended template helpers: https://github.com/influxdata/oapi-codegen/pull/1

### CODE REVIEW

**Review by rogpeppe — COMMENTED** [review]
Thanks for this! I'm afraid I didn't finish reviewing this, but here are some initial thoughts.

File: inluxclient/model/templates/additional-properties.tmpl [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{{range .Types}}{{$addType := .Schema.AdditionalPropertiesType.TypeDecl}}
+
+// Getter for additional properties for {{.TypeName}}. Returns the specified
+// element and whether it was found
+func (a {{.TypeName}}) Get(fieldName string) (value {{$addType}}, found bool) {
+    if a.AdditionalProperties != nil {
+        value, found = a.AdditionalProperties[fieldName]
+    }
+    return
+}
```
[the remark this answers is not in the document]
Comment (vlastahajek) [reply inluxclient/model/templates/additional-properties.tmpl:10]: This template (`additional-properties.tmpl`) and also `constants.tmpl`, `imports.tmpl`, `inline.tmpl` and `typedef.tmpl` are default ones.
 I've changed `client.tmpl`, `client-with-responses.tmpl`, `param-types.tmpl` and `request-bodies.tmpl`.

I guess I should remove default templates from this PR.

Or, do we want to improve all templates?

File: inluxclient/client.go [hunk 2]
```diff
@@ -69,5 +77,26 @@ func New(params Params) (*Client, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error parsing server URL: %w", err)
 	}
+	doer := model.HttpRequestDoer(c.params.HTTPClient)
+	if c.authorization != "" {
+		doer = &authorizationDoer{Client: *c}
+	}
+	c.apiClient, err = model.NewClient(c.apiURL.String(), doer)
+	if err != nil {
+		return nil, fmt.Errorf("error creating server API client: %w", err)
```
[the remark this answers is not in the document]
Comment (vlastahajek) [reply inluxclient/client.go:86]: `%v` is indeed better for non IO functions. IO functions (disk, network) can fail by many reasons which users would like to know by unwrapping error.

**Comment by vlastahajek** [comment]
I've removed the default, not modified, templates

**Review by sranka — COMMENTED** [review]
I am probably missing something, isn't the goal to let the library users use the generated client directly? How can I get the instance of the generated client then? 

Can you also pls rebase the code on top of the latest master, the packages changed a bit.

File: influxclient/model/templates/client.tmpl [hunk 3]
```diff
@@ -0,0 +1,58 @@
+
+// Doer performs HTTP requests.
+//
+// The standard http.Client implements this interface.
+type HTTPRequestDoer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Client which conforms to the OpenAPI3 specification for this service.
+type Client struct {
```
Comment (sranka) [inline influxclient/model/templates/client.tmpl:10]: Shouldn't we rather rename this `Client` to `GeneratedClient` to disabiguate `influxclient/model/Client` and `influxclient/Client`

**Comment by sranka** [comment]
I think that we should also consider relying upon the generated API, we modify templates to output documentation and links to the API doc. [there are some still comments from RogP](https://github.com/influxdata/influxdb-client-go/pull/321#discussion_r839820453) that request improvements in the generated API, additional properties can be simply maps.

**Comment by vlastahajek** [comment]
This PR has become obsolete. The modified templates and generated code are already used in the  Go v2.
The plan is to reuse the management API from Go v2 with a slightly modified API based on the Go V# API design.

**Pull request closed without merging**